ast/expressions_test.go

```diff
@@ -219,3 +219,20 @@ func (tc *testExpressionsSuite) TestDefaultExpr(c *C) {
 	}
 	RunNodeRestoreTest(c, testCases, "insert into t values(%s)", extractNodeFunc)
 }
+
+func (tc *testExpressionsSuite) TestPatternRegexpExprRestore(c *C) {
+	testCases := []NodeRestoreTestCase{
+		{"a regexp 't1'", "`a` REGEXP 't1'"},
+		{"a regexp '^[aeiou]|ok$'", "`a` REGEXP '^[aeiou]|ok$'"},
+		{"a regexp '[0-9a-zA-Z]'", "`a` REGEXP '[0-9a-zA-Z]'"},
+		{"a regexp '^[0-9]{11}$'", "`a` REGEXP '^[0-9]{11}$'"},
+		{"a not regexp 't1'", "`a` NOT REGEXP 't1'"},
+		{"a not regexp '^[aeiou]|ok$'", "`a` NOT REGEXP '^[aeiou]|ok$'"},
+		{"a not regexp '[0-9a-zA-Z]'", "`a` NOT REGEXP '[0-9a-zA-Z]'"},
+		{"a not regexp '^[0-9]{11}$'", "`a` NOT REGEXP '^[0-9]{11}$'"},
```

Comment:
Please also add a test case involving the keywords [`RLIKE`](https://dev.mysql.com/doc/refman/8.0/en/regexp.html#operator_regexp)/`NOT RLIKE`.

Reply:
done.